From 36967603e9a7dd750dae45301e4b5f9d94e6f590 Mon Sep 17 00:00:00 2001
From: Jacopo Mondi <jacopo+renesas@jmondi.org>
Date: Tue, 5 Dec 2017 21:41:48 +0100
Subject: [PATCH 366/909] rcar-vin: rcar-csi2: Don't bail out from probe on no
 ep

commit 9780febd4d1c6a7c0c7b09ca48786c06fe3733b1 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

When rcar-csi interface is not connected to any endpoint, it fails and
bails out from probe before registering its own video subdevice.
This prevents rcar-vin registered notifier from completing and no
subdevice is ever registered, also for other properly connected csi
interfaces.

Fix this not returning an error when no endpoint is connected to a csi
interface and let the driver complete its probe function and register its
own video subdevice.

Signed-off-by: Koji Matsuoka <koji.matsuoka.xm@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/rcar-vin/rcar-csi2.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/rcar-vin/rcar-csi2.c b/drivers/media/platform/rcar-vin/rcar-csi2.c
index 7180f5da7703..422747a61f5f 100644
--- a/drivers/media/platform/rcar-vin/rcar-csi2.c
+++ b/drivers/media/platform/rcar-vin/rcar-csi2.c
@@ -767,8 +767,8 @@ static int rcsi2_parse_dt(struct rcar_csi2 *priv)
 
 	ep = of_graph_get_endpoint_by_regs(priv->dev->of_node, 0, 0);
 	if (!ep) {
-		dev_err(priv->dev, "Not connected to subdevice\n");
-		return -EINVAL;
+		dev_dbg(priv->dev, "Not connected to subdevice\n");
+		return 0;
 	}
 
 	ret = v4l2_fwnode_endpoint_parse(of_fwnode_handle(ep), &v4l2_ep);
-- 
2.17.1

